tensornetwork/linalg/tests/operations_test.py

```diff
@@ -0,0 +1,379 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import numpy as np
+import jax
+import jax.numpy as jnp
+from jax import config
+import tensorflow as tf
+import torch
+import pytest
+import tensornetwork
+import tensornetwork.linalg.operations
+from tensornetwork import backends, backend_contextmanager
+
+#pylint: disable=no-member
+config.update("jax_enable_x64", True)
+
+np_real = [np.float32, np.float16, np.float64]
+np_complex = [np.complex64, np.complex128]
+np_int = [np.int8, np.int16, np.int32, np.int64]
+np_uint = [np.uint8, np.uint16, np.uint32, np.uint64]
+np_float_dtypes = np_real + np_complex
+np_not_bool = np_float_dtypes + np_int + np_uint + [None,]
+np_all_dtypes = np_not_bool + [np.bool,]
+
+torch_supported_dtypes = np_real + np_int + [np.uint8, np.bool, None]
+
+
```

Comment:
It might be worth adding type -tests, i.e. if the array-type of the Tensor and its backend are matching. For example, concatenate seems to violate this (unless Tensor checks this or fixes it)

Reply:
Just saw that `Tensor` converts to tensor, so this will be fine